src/in_mem_filter_store.cpp

```diff
@@ -0,0 +1,506 @@
+            for (uint32_t j = 0; j + 1 < _location_to_labels[i].size(); j++)
+            {
+                label_writer << _location_to_labels[i][j] << ",";
+            }
+            if (_location_to_labels[i].size() != 0)
+                label_writer << _location_to_labels[i][_location_to_labels[i].size() - 1];
+            label_writer << std::endl;
+        }
+        label_writer.close();
+    }
+}
+
+template <typename label_type>
+void InMemFilterStore<label_type>::save_raw_labels(const std::string &save_path, const size_t total_points)
+{
+    if (_label_map.empty())
+    {
+        diskann::cout << "Warning: not saving raw labels as label map is empty" << std::endl;
+        return;
+    }
+    std::unordered_map<label_type, std::string> mapped_to_raw_labels;
+    // invert label map
+    for (const auto &[key, value] : _label_map)
+    {
+        mapped_to_raw_labels.insert({value, key});
+    }
+
+    // write updated labels
+    std::ofstream raw_label_writer(save_path);
+    assert(raw_label_writer.is_open());
+    for (uint32_t i = 0; i < total_points; i++)
+    {
+        for (uint32_t j = 0; j + 1 < _location_to_labels[i].size(); j++)
+        {
+            raw_label_writer << mapped_to_raw_labels[_location_to_labels[i][j]] << ",";
+        }
+        if (_location_to_labels[i].size() != 0)
+            raw_label_writer << mapped_to_raw_labels[_location_to_labels[i][_location_to_labels[i].size() - 1]];
+
+        raw_label_writer << std::endl;
+    }
+    raw_label_writer.close();
+}
+
+template <typename label_type> void InMemFilterStore<label_type>::save_universal_label(const std::string &save_path)
+{
+    if (_has_universal_label)
+    {
+        std::ofstream universal_label_writer(save_path);
+        assert(universal_label_writer.is_open());
+        // universal_label_writer << _universal_label << std::endl;
+
+        universal_label_writer << _universal_label << std::endl;
+        universal_label_writer.close();
+    }
+}
+
+template <typename label_type> void InMemFilterStore<label_type>::save_medoids(const std::string &save_path)
+{
+    if (_label_to_medoid_id.size() > 0)
+    {
+        std::ofstream medoid_writer(save_path);
+        if (medoid_writer.fail())
+        {
+            throw diskann::ANNException(std::string("Failed to open medoid file ") + save_path, -1);
+        }
+        for (auto iter : _label_to_medoid_id)
+        {
+            medoid_writer << iter.first << ", " << iter.second << std::endl;
+        }
+        medoid_writer.close();
+    }
+}
+
+template <typename label_type> void InMemFilterStore<label_type>::save_label_map(const std::string &save_path)
+{
+    if (_label_map.empty())
+    {
+        diskann::cout << "Warning: not saving label map as it is empty." << std::endl;
+        return;
+    }
+    std::ofstream map_writer(save_path);
+    for (auto mp : _label_map)
+    {
+        map_writer << mp.first << "\t" << mp.second << std::endl;
+    }
+    map_writer.close();
+}
+
```

Comment:
we need a uber-method called save which can call the necessary saves the store requires like label map, etc.